.github/workflows/static_analysis.yaml

```diff
@@ -13,7 +13,7 @@ jobs:
     name: cppcheck
     runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 #v3
```

Comment:
I think we can use checkout v4 here with the hash: 692973e3d937129bcbf40652eb9f2f61becf3332
as we are already using v4 in other files..